core/src/main/scala/org/http4s/headers/Content-Length.scala

```diff
@@ -12,18 +12,24 @@ import org.http4s.util.Writer
   *
   * @param length the length; throws an `IllegalArgumentException` if negative
   */
-final case class `Content-Length`(length: Long) extends Header.Parsed {
-  require(length >= 0L)
-
+sealed abstract case class `Content-Length`(length: Long) extends Header.Parsed {
```

Comment:
That's a good idea, i'll switch to that construct instead of the anonymous classes